packages/grid/x-data-grid-pro/src/tests/exportPrint.DataGrid.test.tsx

```diff
@@ -0,0 +1,262 @@
+                  />
+                ),
+              }}
+              initialState={{
+                columns: {
+                  columnVisibilityModel: {
+                    // Hide columns status and traderName, the other columns will remain visible
+                    col1: gridVisible,
+                    col2: false,
+                  },
+                },
+              }}
+            />
+          </div>,
+        );
+        expect(onColumnVisibilityModelChange.callCount).to.equal(0);
+
+        // Trigger print
+        fireEvent.click(screen.queryByRole('button', { name: 'Export' }));
+        fireEvent.click(screen.queryByRole('menuitem', { name: 'Print' }));
+        await clock.runToLast();
+        expect(onColumnVisibilityModelChange.callCount).to.equal(2);
+        // verify column visibility has been set
+        expect(onColumnVisibilityModelChange.firstCall.firstArg).to.deep.equal({
+          col1: printVisible,
+          col2: true,
+        });
+
+        // verify column visibility has been restored
+        expect(onColumnVisibilityModelChange.secondCall.firstArg).to.deep.equal({
+          col1: gridVisible,
+          col2: false,
+        });
+      });
+
+      it(`should set and restore 'col1' from ${printVisible ? "'visible'" : "'hidden'"} to ${
+        gridVisible ? "'visible'" : "'hidden'"
+      } (hidden)`, async function test() {
+        const onColumnVisibilityModelChange = spy();
+
+        render(
+          <div style={{ width: 300, height: 300 }}>
+            <DataGrid
+              {...baselineProps}
+              onColumnVisibilityModelChange={onColumnVisibilityModelChange}
+              components={{
+                Toolbar: () => (
+                  <GridToolbarExport
+                    printOptions={{ fields: printVisible ? ['col1', 'col2'] : ['col2'] }}
+                  />
+                ),
+              }}
+              columns={[
+                { field: 'col1', hide: !gridVisible },
+                { field: 'col2', hide: true },
+              ]}
+            />
+          </div>,
+        );
+        expect(onColumnVisibilityModelChange.callCount).to.equal(0);
+
+        // Trigger print
+        fireEvent.click(screen.queryByRole('button', { name: 'Export' }));
+        fireEvent.click(screen.queryByRole('menuitem', { name: 'Print' }));
+        await clock.runToLast();
+
+        expect(onColumnVisibilityModelChange.callCount).to.equal(2);
+
+        // verify column visibility has been set
+        expect(onColumnVisibilityModelChange.firstCall.firstArg).to.deep.equal({
+          col1: printVisible,
+          col2: true,
+        });
+
+        // verify column visibility has been restored
+        expect(onColumnVisibilityModelChange.secondCall.firstArg).to.deep.equal({
+          col1: gridVisible,
+          col2: false,
+        });
+      });
+    });
+  });
+  describe('column selection', () => {
```

Comment:
"selection" remembers me of checkbox selection.

```suggestion

  describe('columns to print', () => {
```